cmd/containerd-shim-runhcs-v1/events.go

```diff
@@ -18,27 +21,37 @@ var _ = (publisher)(publishEvent)
 var publishLock sync.Mutex
 
 func publishEvent(ctx context.Context, topic string, event interface{}) {
-	_, span := trace.StartSpan(ctx, "publishEvent")
+	publishEventInner(ctx, topic, event)
+}
+
+func publishEventInner(ctx context.Context, topic string, event interface{}) (err error) {
```

Comment:
Can you just declare `var err error`  above the `defer func() { oc.SetSpanStatus... ` and then you dont need to have the `publishEventInner`

Reply:
I could, and did before I changed it to this. I thought this way was more idiomatic. I can change it back if you prefer.

Reply:
If we ever did want to return the error back up to the caller, we don't have to make many changes this way.

Reply:
Can you look how I did it here. I think we should just do this: https://github.com/microsoft/hcsshim/pull/645/files#diff-a5aed6d593a0f625adad2409ebb826ebR21

Reply:
Isn't it better to set the span status to failed in the event of an error. Otherwise it will appear the `publishEvent` span succeeded but there were actually errors in the publishing.

Reply:
It seems weird to me to explicitly log an error at each exit point instead of just treating this like a normal function that returns an error.

Reply:
Can you just change the func to return the error then. I don't want to have this weird inner / outer function. Callers can just ignore the error for now. Then you can just do a normal defer `oc.SetSpanStatus`

Reply:
Sure, just wasn't sure how important keeping the `publisher` type the same was. I've updated with this change.